packages/web/src/scss/components/Modal/README.md

```diff
@@ -282,47 +232,89 @@ You can still close modal with close buttons or ESC key.
   </div>
-  <div class="ScrollView__overflowDecorators ScrollView__overflowDecorators--borders" aria-hidden="true"></div>
-</div>
+  <!-- ModalFooter -->
+</article>
 ```
 
-### Disable Scrolling Inside ModalDialog
+### Custom Height
+
+By default, ModalDialog grows to the height of its content until it reaches the [maximum height](#custom-max-height)
+limit.
 
-Scrolling inside ModalDialog can be turned off by adding the `ModalDialog--nonScrollable` modifier class:
+You can set a custom preferred height of ModalDialog using a custom property:
+
+- `--modal-preferred-height-mobile` for mobile screens, and
+- `--modal-preferred-height-tablet` for tablet screens and up.
+
+This is useful for Modals with dynamic content, e.g. a list of items that can be added or removed, or a multistep wizard.
 
 ```html
-<article class="ModalDialog ModalDialog--nonScrollable">
+<article
+  class="ModalDialog ModalDialog--scrollable"
+  style="--modal-preferred-height-mobile: 400px; --modal-preferred-height-tablet: 500px;"
+>
   <!-- … -->
 </article>
 ```
 
-This way, the ModalBody will expand to fit the height of its content and the whole ModalDialog will scroll in case the
-content is longer than user's viewport.
+👉 This feature is only available for ModalDialogs with the `ModalDialog--scrollable` modifier class.
+
+👉 Please note the custom height values are considered **preferred**. Scrollable ModalDialog will never expand beyond
+the viewport height. See the [Custom Max Height](#custom-max-height) section for more information.
 
-👉 Please note that this modifier class can produce unexpected results when used in combination with ScrollView.
+### Custom Max Height
+
+The default maximum height of a scrollable ModalDialog is **600 px**, as long as it can fit the viewport.
 
-#### ⚠️ DEPRECATION NOTICE
+If the viewport is smaller, scrollable ModalDialog will shrink to fit the viewport. In such case, the ModalDialog height
+will calculate as "viewport height minus 1100 spacing" (which can be translated as `100dvh - 32px`).
 
-The `.ModalDialog--nonScrollable` modifier will be removed in the next major release and the ModalDialog will be made
-non-scrollable by default. It will be possible to re-enable the inside scrolling by adding the
-`.ModalDialog--scrollable` modifier class (which will remain the recommended default usage).
+You can use the custom property `--modal-max-height-tablet` to override the default maximum height limit on tablet
+screens and up:
+
+```html
+<article class="ModalDialog ModalDialog--scrollable" style="--modal-max-height-tablet: 400px">
+  <!-- … -->
+</article>
+```
+
+👉 This feature is only available for ModalDialogs with the `ModalDialog--scrollable` modifier class.
+
+👉 If a [custom height](#custom-height) is set, the custom max height is only applied if it's smaller than the custom
+height.
+
+👉 Please note the max height on mobile screens is currently not customizable. Let us know if you need this feature! 🙏
```

Comment:
But the second issue will add this, right?

Reply:
Deleted in advance.